doc/scss/custom.scss

```diff
@@ -190,3 +190,54 @@ div.sk-text-image-grid-small {
 div.sk-text-image-grid-large {
   @include sk-text-image-grid(100px);
 }
+
+/* Style of the docsearch modal window */
+
+// In the following, we mainly reuse the background color and font-color from the
+// pydata-sphinx-theme. We also adjust slightly the font-size to be wider and in line
+// with the sidebar design.
+:root {
+  --docsearch-modal-width: 768px;
+  --docsearch-modal-height: 700px;
+  --docsearch-searchbox-shadow: inset 0 0 0 2px var(--pst-color-primary);
+  --docsearch-highlight-color: var(--pst-color-primary);
+}
+
+.DocSearch-Modal {
+  background: var(--pst-color-background);
+}
+
+.DocSearch-Form {
+  background: var(--pst-color-background);
+}
+
+.DocSearch-MagnifierLabel,
+.DocSearch-Reset {
+  color: var(--pst-color-primary);
+}
+
+.DocSearch-Input {
+  color: var(--pst-color-text-base);
+  font-size: var(--pst-font-size-icon);
+}
+
```

Comment:
The icons are misaligned due to the styling on svg.

```scss
.DocSearch-Hit-Icon svg {
  vertical-align: unset;
}
```

Before:

![image](https://github.com/scikit-learn/scikit-learn/assets/108576690/15e0cc20-c06d-4d44-82ab-fab8a4317183)

After:

![image](https://github.com/scikit-learn/scikit-learn/assets/108576690/3b11d7c1-b942-47a1-91e7-a0704c36c9ae)